Libraries/LibELF/DynamicLoader.h

```diff
@@ -117,11 +114,15 @@ class DynamicLoader : public RefCounted<DynamicLoader> {
     void load_program_headers();
 
     // Stage 2
-    void do_relocations(size_t total_tls_size);
+    void do_main_relocations(size_t total_tls_size);
+
+    // Stage 3
+    void do_lazy_relocations(size_t total_tls_size);
     void setup_plt_trampoline();
     void call_object_init_functions();
 
     bool validate();
+    void do_relocation(size_t total_tls_size, DynamicObject::Relocation relocation, Vector<DynamicObject::Relocation>& lazy_relocation_queue);
```

Comment:
Passing a vector of unresolved relocations feels inconvenient.
Maybe we could instead have `do_relocation` return an Enum type that reports whether the we succeeded to perform the relocation / failed / failed on a weak symbol ?
Then callers to the function could add the symbol to the lazy relocation list if they wish to do so.